src/dlsproto/node/neo/request/GetRange.d

```diff
@@ -12,13 +12,527 @@
+            {
+                payload.addConstant(RequestStatusCode.Started);
+            }
+        );
+
+        this.value_buffer = this.resources.getVoidBuffer();
+        this.batch_buffer = this.resources.getVoidBuffer();
+        this.compressed_batch = this.resources.getVoidBuffer();
+        this.saved_exception = this.resources.getException();
+        this.lzo = this.resources.getLzo();
+
+        try
+        {
+            this.run();
+        }
+        finally
+        {
+            this.requestFinished();
+        }
+    }
+
+    /***************************************************************************
+
+        Request main loop. Gets records from the storage, sends them to the
+        client in batches and handles Continue/Stop client feedback.
+
+    ***************************************************************************/
+
+    private void run ()
+    {
+    main_loop: while (true)
+        {
+            time_t record_timestamp;
+
+            bool wait_for_data;
+            auto got_next = this.getNextRecord(record_timestamp, *this.value_buffer,
+                    wait_for_data);
+
+            if (wait_for_data)
+            {
+                auto event = this.ed.nextEvent(ed.NextEventFlags.Receive);
+
+                switch (event.active)
+                {
+                    // Resumed by the storage engine, now ready to return value
+                    // with `getNextRecord`, so we try again.
+                    case event.active.resumed:
+                        continue main_loop;
+
+                    // received from the client
+                    case event.active.received:
+                        // This should be stop. Acknowledge it exit
+                        this.verifyReceivedMessageIsStop(event.received.payload);
+                        this.sendStoppedMessage();
+                        return;
+
+                    default:
+                        throw this.ed.shutdownWithProtocolError(
+                            "GetRange: Unexpected fiber resume code");
+                }
+            }
+
+            if(!got_next)
+            {
+                this.sendBatchAndReceiveFeedback();
+
+                this.ed.nextEvent(
+                    this.ed.NextEventFlags.Receive,
+                    (ed.Payload payload)
+                    {
+                        payload.addConstant(MessageType.Finished);
+                    }
+                );
+
+                bool acked;
+                do
+                {
+                    auto event = this.ed.nextEvent(
+                            this.ed.NextEventFlags.Receive);
+
+                    if (event.active.received)
+                    {
+                        MessageType msg_type;
+                        this.parser.parseBody(event.received.payload, msg_type);
+
+                        if (msg_type == msg_type.Ack)
+                        {
+                            acked = true;
```

Comment:
Suggest replacing this variable with a `while (true)` loop and `break` here.

Reply:
To be honest, I find the current approach more verbose (in a good way). I can still change it, if you insist.